src/hotspot/share/compiler/methodLiveness.cpp

```diff
@@ -618,18 +618,22 @@ void MethodLiveness::BasicBlock::compute_gen_kill_single(ciBytecodeStream *instr
       // These bytecodes have no effect on the method's locals.
       break;
 
-    case Bytecodes::_return:
-      if (instruction->method()->intrinsic_id() == vmIntrinsics::_Object_init ||
-          (instruction->method()->is_object_constructor() && instruction->method()->holder()->is_inlinetype())) {
-        // return from Object.init implicitly registers a finalizer
-        // for the receiver if needed, so keep it alive.
-        // Value class constructors update the scalarized receiver. Keep it live so that
-        // we can find it after (chained) constructor calls and propagate updates to the caller.
+    case Bytecodes::_return: {
+      ciMethod* method = instruction->method();
+      ciInstanceKlass* holder = method->holder();
+      if (method->intrinsic_id() == vmIntrinsics::_Object_init ||
+          (method->is_object_constructor() &&
+           ((!holder->is_abstract() && holder->is_inlinetype()) ||
+            holder->is_abstract()))) {
```

Comment:
We now keep more receivers live that might not be needed (non-value abstract classes). But I think this is the simplest/most straight forward solution since we do not have access to the actual receiver node here to check its bottom type as done in the code during parsing.